static/js/ui_init.js

```diff
@@ -34,7 +34,7 @@ function message_hover(message_row) {
     message_row.addClass('message_hovered');
     current_message_hover = message_row;
 
-    if (!message.sent_by_me) {
+    if (!message.sent_by_me && !page_params.realm_allow_community_topic_editing) {
```

Comment:
I think this block needs to actually use `get_editability` so that if it's past the time limit, this doesn't appear.